extensions/2.0/Khronos/KHR_materials_transmission/README.md

```diff
@@ -0,0 +1,138 @@
+
+This extension aims to address the simplest and most common use cases for optical transparency: infinitely-thin materials with no complex scattering (e.g. dynamic scattering) or dispersion. Dealing exclusively with “thin” materials (i.e. materials where only the surface is considered and not the volume) allows many simplifications to be made when calculating things like refraction and absorption.
+
+## Extending Materials
+
+A transparent material is defined by adding the `KHR_materials_transmission` extension to any glTF material. When present, the extension indicates that a material should be rendered as a transparent surface and be blended as defined in the following spec. Alpha coverage and `doubleSided` properties still apply to transparent materials. Note that, as rendering transparent objects presents many difficult-to-solve issues with primitive ordering, this extension does not dictate rendering algorithms.
+
+```json
+materials: [
+  {
+    "extensions": {
+       "KHR_materials_transmission": {
+         "transmissionFactor": 0.8,
+         "transmissionTexture": 0
+       }
+    }
+  }
+]
+```
+
+## Properties 
+
+Only two properties are introduced with this extension and combine to describe a single value; the percentage of light that is transmitted through the surface of the material. These properties work together with the existing properties of the material to define the way light is modified as it passes through the substance. 
+
+|   |Type|Description|Required|
+|---|----|-----------|--------|
+|**transmissionFactor** | `number` | The base percentage of light that is transmitted through the surface.| No, default:`1.0`|
+|**transmissionTexture** | [`textureInfo`] | A greyscale texture that defines the transmission percentage of the surface. This will be multiplied by transmissionFactor. | No |
+
+### transmissionFactor 
+
+The amount of light that is transmitted by the surface rather than diffusely re-emitted. This is a percentage of all the light that penetrates a surface (i.e. isn’t specularly reflected) rather than a percentage of the total light that hits a surface. A value of 1.0 means that 100% of the light that penetrates the surface is transmitted through. This is the default.
+
+<figure>
+  <img src="./figures/ConstantTransmission.png"/>
+<figcaption><em>Defaulting to 1.0 allows making a material transparent by simple defining the extension.</em></figcaption>
+</figure>
+
+### transmissionTexture 
+
+A greyscale texture that defines the amount of light that is transmitted by the surface rather than absorbed and re-emitted. A value of 1.0 means that 100% of the light that penetrates the surface (i.e. isn’t specularly reflected) is transmitted through. The value is linear and is multiplied by the transmissionFactor to determine the total transmission value. Note that use of transmission values does not disallow usage of alpha coverage (via `baseColor.a`).
```

Comment:
Could it be a single channel selected for this? If it's red, we could pair it with the metallic roughness texture.

Reply:
The red channel of the metallic-roughness texture can already be used for ambient occlusion.
However, I had thought that we could share this texture with the KHR_materials_volume (#1726) extension which could use one of the other channels for thickness information (if we go the route of allowing the thickness param).

Reply:
Dispersion (the prism effect) is specifically not part of this extension to keep things simple. It could certainly be included in another extension though.

Reply:
I agree with @jackcaron that we should pick a single channel, for two reasons:

1. By requiring that three channels be filled with the same value, we are losing any packing flexibility that might (intentionally or not) have been available. The fact that the <kbd>r</kbd> channel in the ORM texture could be used for either AO *or* transmission is a good thing.
2. If in the future we have a general-purpose swizzle extension, it will be awkward that it's required to provide a 3-channel output for a 1-channel value here.

Either <kbd>r</kbd> or <kbd>a</kbd> seem like reasonable choices, to me.

Reply:
I would propose this extension use the r channel of a texture separate from the ORM, as I would not want to have to omit occlusion in order to take advantage of per-pixel transmission. Pre-calculated ambient occlusion has a strong visual benefit.

Perhaps we could use a channel of another texture, and reserve the green and blue channels for other PBR Next features? Transmission (r), clearcoatTexture (g), and clearcoatRoughnessTexture (b) for example.

Reply:
We don't need to choose a specific texture to pair transmission with — as long as the channels don't conflict, users can mix and match. Any of (transmission; occlusion/rough/metal), (occlusion; transmission/rough/metal), (transmission; occlusion; ____/rough/metal) would be allowed.

Reply:
We've been discussing using the 'G' channel for thickness over in the [KHR_materials_volume](https://github.com/KhronosGroup/glTF/pull/1726) thread.

Reply:
I agree that the texture should be one channel (prism not being part of this extension).
As for which channel to choose I would propose either R or A.
Picking A would make it easy to have occlusion/roughness/metallic/transmission (without swizzle)
Picking R could have benefit if occlusion is less likely to be used with this extension.

Either way is fine to me, I will solve by internally swizzling the textures if needed.